src/domain/contracts/contracts.repository.ts

```diff
@@ -19,4 +34,50 @@ export class ContractsRepository implements IContractsRepository {
     const data = await api.getContract(args.contractAddress);
     return ContractSchema.parse(data);
   }
+
+  async isTrustedForDelegateCall(args: {
+    chainId: string;
+    contractAddress: `0x${string}`;
+  }): Promise<boolean> {
+    return this.isTrustedForDelegateCallContractsListEnabled
+      ? await this.isIncludedInTrustedForDelegateCallContractsList({
+          chainId: args.chainId,
+          contractAddress: args.contractAddress,
+        })
+      : await this.isTrustedForDelegateCallContract({
+          chainId: args.chainId,
+          contractAddress: args.contractAddress,
+        });
+  }
+
+  private async getTrustedForDelegateCallContracts(
+    chainId: string,
+  ): Promise<Page<Contract>> {
+    const api = await this.transactionApiManager.getApi(chainId);
+    const contracts = await api.getTrustedForDelegateCallContracts();
+    return ContractPageSchema.parse(contracts);
+  }
+
+  private async isIncludedInTrustedForDelegateCallContractsList(args: {
+    chainId: string;
+    contractAddress: `0x${string}`;
+  }): Promise<boolean> {
+    const trustedContracts = await this.getTrustedForDelegateCallContracts(
+      args.chainId,
+    );
+    return trustedContracts.results.some((contract) =>
+      isAddressEqual(contract.address, args.contractAddress),
```

Comment:
Althought theoretically correct, I think we should still check the flag to be safe.
```suggestion
      isAddressEqual(contract.address, args.contractAddress) &&
        contract.trustedForDelegateCall,```

Reply:
Agreed. Even if it's not needed theoretically, we can check the flag to avoid the case the filter is not working properly on the Transaction Service for some reason. Thanks. Changed in [7e3b393](https://github.com/safe-global/safe-client-gateway/pull/2430/commits/7e3b393d2742e73aa195cc9f47298be406c860ad)

Reply:
Should we want to expose pages of contracts later on, we are also covered, e.g. we call a hypothetical `getContracts` method on the domain, passing the query there instead.